pkg/obfuscator/domain.go

```diff
@@ -54,13 +58,18 @@ func (d *domainObfuscator) obfuscatedDomain(domain string) string {
 	if replacement, ok := d.domainMapping[domain]; ok {
 		return replacement
 	}
-	d.domainCount++
-	replacement := fmt.Sprintf(obfuscatedTemplate, d.domainCount)
+	var replacement string
+	switch d.replacementType {
+	case schema.ObfuscateReplacementTypeStatic:
+		replacement = d.obfsGenerator.generateStaticReplacement()
+	case schema.ObfuscateReplacementTypeConsistent:
+		replacement = d.obfsGenerator.generateConsistentReplacement()
+	}
```

Comment:
it's now the third time I find that exact same switch statement, can we pull that into a dedicated struct+interface along with the replacementType as input and refactor the other obfuscators towards that as well?

Reply:
Yeah, I will address this.